docs/variables.md

```diff
@@ -184,6 +184,16 @@ Only available for `successCondition`
 | `response.body` | Response body (`string`) |
 | `response.headers` | Response headers (`map[string][]string`) |
 
+### CronWorkflows
```

Comment:
I was expecting to be able to access namespace, annotations and labels like you can from `workflow`.

I don't know what for in this scenario, but just as general variables which would be in scope to mirror workflows

Reply:
I might suggest leaving those out for now until there's more of a need from users (as I did with some of the previous ones Isitha now removed). Especially since the variables are mostly useful for `when` and we recommend using `workflowTemplateRef`, so there isn't much of a place to use them either

Reply:
I know I wrote "I don't know what for" - but actually now my use case would be the "minimum interval" one - I'd write an identical expression for `when` on all workflows, which pulled minimum interval from an annotation.

Reply:
That is a clever implementation, though I'd hazard a guess that it's not one that a majority of users would use 😅 
So "more of a need from users" would still apply

Reply:
@agilgur5 I personally see no harm in including labels and annotations, I would argue that I would actually expect them there since workflows has them.

Reply:
> I would argue that I would actually expect them there since workflows has them.

I'm not sure that the `govaluate` version of `when` actually has this (though you can throw an `expr` expression anyway)

> without any negative consequences

Every feature is more things to maintain, have backward compat for, etc

Reply:
> That is a clever implementation, though I'd hazard a guess that it's not one that a majority of users would use 😅 So "more of a need from users" would still apply

This PR is addressing a user need, and we'd be suggesting this mechanism for them to implement their requirements if it were available :shrug:. So we'd like it to be available.

The alternative for another use of this when clause would be datetime literals in the when clause (more than once due to wanting to match day+month+year, which is also icky.

Reply:
> It feels like a weird asymmetry, I'm not sure why the variable scopes should look different for the two types where they don't need to.

There will be asymmetry already; there are [several other variables available on Workflows](https://argo-workflows.readthedocs.io/en/release-3.5/variables/#global), so really the question is where to draw the line, not whether to make them entirely symmetrical.

> The alternative for another use of this when clause would be datetime literals in the when clause (more than once due to wanting to match day+month+year, which is also icky).

Doesn't the annotation approach also require a literal? I'm confused. You're only replacing the `interval` variable.
Or do you mean a different use-case entirely?
Also `expr` fortunately does have [`date` manipulation functions](https://expr-lang.org/docs/language-definition#date)

> and we'd be suggesting this mechanism for them to implement their requirements if it were available 🤷. So we'd like it to be available.

You'd be suggesting annotations for them? I guess that depends on the experience level of the user 🤔 ; annotations and template substitution can be pretty complicated for beginners to grasp, but for more advanced users that would make sense I suppose

But yea, if you'd be suggesting that, I suppose that's enough need. The point on symmetry still stands though

Reply:
This would be something we'd be designing, and the user would just fill out the annotations.

Reply:
Oh. Like a pre-filled template?
Huh I hadn't even thought of that tbh. That makes a lot more sense 😅 Thanks for elaborating that!

I see, CronWorkflows themselves don't have parameter input so you'd need to meta-template it. I'd probably personally end up doing that with Helm rather than raw (actually talked about this with Isitha last week, I did a lot of Helm -> WorkflowTemplate -> submit Workflow through API -> create CD Application -> Helm -> WorkflowTemplate at my last job, so it was _very_ meta. Understandably, that's not the route for everyone 😅)